src/groupWith.js

```diff
@@ -0,0 +1,33 @@
+var _curry2 = require('./internal/_curry2');
+
+/**
+ * Takes a list and returns a list of lists where each sublist contains only equal elements.
+ *
+ * @func
+ * @memberOf R
+ * @category List
+ * @sig (a, a -> Boolean) -> [a] -> [[a]]
+ * @param {Function} fn Function :: a, a -> Boolean
+ * @param {Array} list The array to group
+ * @return {List} A list that contains sublists of equal elements,
+ *         whose concatenations is equal to the original list.
+ * @example
+ *
+ *      var groupWithLetters = R.groupWith(R.equals);
+ *      var letters = ["M", "i", "s", "s", "i", "s", "s", "i", "p", "p", "i"];
+ *      groupWithLetters(letters);
+ *      // [['M'], ['i'], ['s', 's'], ['i'], ['s', 's'], ['i'], ['p', 'p'], ['i']]
```

Comment:
The problem I have with *only* using `R.equals` is that at a glance the function appears to be specialized for grouping contiguous equal values.

Let's have our :cake: and eat it too!

```javascript
> groupWith(R.equals, [0, 1, 1, 2, 3, 5, 8, 13, 21])
[[0], [1, 1], [2, 3, 5, 8, 13, 21]]

> groupWith((a, b) => a % 2 === b % 2, [0, 1, 1, 2, 3, 5, 8, 13, 21])
[[0], [1, 1], [2], [3, 5], [8], [13, 21]]
```